server/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -531,7 +532,10 @@ public Script(ScriptType type, String lang, String idOrCode, Map<String, String>
 
         if (type == ScriptType.INLINE) {
             this.lang = Objects.requireNonNull(lang);
-            this.options = Collections.unmodifiableMap(Objects.requireNonNull(options));
+            // making sure if content_type was application/json, application/json; charset=UTF-8 (note space and casing)
+            // it will be treated as the XContentType.JSON.mediaType() -> application/json;charset=utf-8
+            this.options = Maps.copyMapWithModifiedEntryWhenPresent(Collections.unmodifiableMap(options), CONTENT_TYPE_OPTION,
```

Comment:
we don't document much about this `content_type` field in scripts. I suspect users might still send the `application/json; charset=UTF-8`. This line is meant to make it backwards compatible with that value and always set `application/json;charset=utf-8` as an option when `application/json*` was used.